content/blog/simplify-embedded-projects-with-esp-bsp/index.md

```diff
@@ -0,0 +1,210 @@
+    dependencies:
+        espressif/m5stack_core_s3: "^1.1.0"
+        esp_jpeg: "^1.0.5~2"
+        esp_codec_dev:
+            public: true
+            version: "==1.1.0"
+    ## Required IDF version
+    idf:
+        version: ">=5.0.0"
+    ```
+
+4. **Build and Flash the Project**:
+
+   Compile and flash your application to the ESP32-S3-BOX-3:
+
+   ```bash
+   idf.py build flash monitor
+   ```
+
+   Note: Use `Ctrl+]` to quit the monitor application.
+
+   Or, you can try it directly with ESP Launchpad and flash precompiled binary to your ESP32-S3-BOX-3:
+
+   {{< button href="https://espressif.github.io/esp-launchpad/?flashConfigURL=https://espressif.github.io/esp-bsp/config.toml&app=display_audio_photo" target="_blank" >}}
+     Try it with ESP Launchpad
+   {{< /button >}}
+
+
+   Note: Some models of M5Stack CoreS3 have QUAD SPI RAM, in that case please use the following command for the build, which will take into account specific settings for M5Stack CoreS3 from the file:
+
+   ```bash
+   idf.py build -DSDKCONFIG=sdkconfig.bsp.m5stack_core_s3
+   ```
+
+### Exploring the Example
+
+Once the application is running, you’ll see the following features in action:
+
+- **Display**: Shows images, text files, and more.
+- **Touch**: Interacts with the display.
+- **Audio**: Plays sound files.
+
+Let's look at the source code of the example. Board Support Package provides API which allows to initialize a board. The application code the can be cleaner and does not need to require board specific details like pins definition.
+
+```c
+#include "esp_log.h"
+#include "bsp/esp-bsp.h"
+#include "app_disp_fs.h"
+
+static const char *TAG = "example";
+
+void app_main(void)
+{
+    /* Initialize and mount SPIFFS */
+    bsp_spiffs_mount();
+
+    /* Initialize I2C (for touch and audio) */
+    bsp_i2c_init();
+
+    /* Initialize display and LVGL */
+    bsp_display_start();
+
+    /* Set default display brightness */
+    bsp_display_brightness_set(APP_DISP_DEFAULT_BRIGHTNESS);
+
+    /* Add and show LVGL objects on display */
+    app_disp_lvgl_show();
+
+    /* Initialize SPI flash file system and show list of files on display */
+    app_disp_fs_init();
+
+    /* Initialize audio */
+    app_audio_init();
+
+    ESP_LOGI(TAG, "Example initialization done.");
+}
+```
+
+### Simulating the ESP32-S3-BOX-3 with Wokwi
+
+You can also simulate the application with ESP-BSP for ESP32-S3-BOX-3 using the Wokwi simulator from your development environment using a plugin for [VS Code](https://docs.wokwi.com/vscode/getting-started), [Espressif IDE](https://github.com/espressif/idf-eclipse-plugin?tab=readme-ov-file#wokwi-simulator), or [JetBrains IDEs](https://plugins.jetbrains.com/plugin/23826-wokwi-simulator). This allows you to test your applications without needing the physical hardware.
+
+To set up the simulation, you need to add two files to your project: [wokwi.toml](wokwi.toml) and [diagram.json](diagram.json) in the root of the project.
```

Comment:
Downloading code files directly from a website to include in your project feels like a workaround for lack of a better choice these days. At least, I went through all kinds of tutorials myself in recent months and do not remember any cases where I had to download code files directly from an article.

Can these files be stored in a repo? Or maybe it would be better to give them in code blocks in the article directly?

Reply:
True, makes sense. Added into a content, those files are not too big.